File: mciv/src/main/java/com/mordrum/mciv/common/Civilization.java
package com.mordrum.mciv.common;

import java.util.List;
import java.util.UUID;

public class Civilization {
	private final long id;
	private final String name;
	private final List<UUID> players;
	private boolean inviteOnly;
	private final String bannerId;

	public Civilization(long id, String name, String bannerId, boolean inviteOnly, List<UUID> players) {
		this.id = id;
		this.name = name;
		this.players = players;
		this.inviteOnly = inviteOnly;
		this.bannerId = bannerId;
	}

	public long getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public List<UUID> getPlayers() {
		return players;
	}

	public boolean isInviteOnly() {
		return inviteOnly;
	}

	public void setInviteOnly(boolean inviteOnly) {
		this.inviteOnly = inviteOnly;
	}

	public String getBanner() {
		return bannerId;
	}
}

File: mciv/src/main/java/com/mordrum/mciv/common/models/Civilization.kt
package com.mordrum.mciv.common.models

import com.mordrum.mcore.client.gui.ChatColor
import java.util.*

class Civilization : SuperModel() {
    var name: String = ""
    var banner: String = ""
    var players: List<Player> = ArrayList()
    var inviteOnly: Boolean = false

    var founder: Player? = null

    var chunks: List<CivilizationChunk> = ArrayList()

    var primaryColor: ChatColor = ChatColor.YELLOW
    var secondaryColor: ChatColor = ChatColor.BLUE
    var tertiaryColor: ChatColor = ChatColor.GREEN
    // Short version of the civ name, shown in chat and other places
    var tag = ""
    // Shown when a non-citizen enters this civ's land
    var welcomeMessage = ""
    // Shown to all citizens upon login
    var motd = ""
    // Shown to anyone who looks at the civilization in a menu of some sort
    var description = ""
    // Allied civilizations
    var allies: List<Civilization> = ArrayList()
    // Civilizations this civ is at war with
    var enemies: List<Civilization> = ArrayList()
    // What this civ focuses on primarily
    var primaryFocus: Focus? = null
    var secondaryFocuses: Collection<Focus> = ArrayList()

    enum class Focus {
        MILITARY, TRADING, PRODUCTION, SCIENCE, EXPANSION
    }
}